Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/ControllerPoseSynchronizer.cs

```diff
@@ -91,11 +93,16 @@ public MixedRealityInputAction PoseAction
         #region IMixedRealitySourcePoseHandler Implementation
 
         /// <inheritdoc />
-        public virtual void OnSourceDetected(SourceStateEventData eventData) { }
+        public virtual void OnSourceDetected(SourceStateEventData eventData)
+        {
+            hasSource = true;
+        }
 
         /// <inheritdoc />
         public virtual void OnSourceLost(SourceStateEventData eventData)
         {
+            hasSource = false;
```

Comment:
This event will fire for all lost sources, not just the one this pose synchronizer is listening for. This should probably be inside the below `if` statement (pending my first comment, though).